docs/Validation.md

```diff
@@ -320,42 +320,46 @@ const CustomerCreate = () => (
 
 ## Server-Side Validation
 
-You can use the errors returned by the dataProvider mutation as a source for the validation. In order to display the validation errors, a custom `save` function needs to be used:
+Server-side validation is supported out of the box. It requires that the dataProvider throws an error with the following shape:
 
-{% raw %}
-```jsx
-import * as React from 'react';
-import { useCallback } from 'react';
-import { Create, SimpleForm, TextInput, useCreate } from 'react-admin';
-
-export const UserCreate = () => {
-    const [create] = useCreate();
-    const save = useCallback(
-        async values => {
-            try {
-                await create('users', { data: values }, { returnPromise: true });
-            } catch (error) {
-                if (error.body.errors) {
-                    // The shape of the returned validation errors must match the shape of the form
-                    return error.body.errors;
-                }
-            }
-        },
-        [create]
-    );
-
-    return (
-        <Create>
-            <SimpleForm onSubmit={save}>
-                <TextInput label="First Name" source="firstName" />
-                <TextInput label="Age" source="age" />
-            </SimpleForm>
-        </Create>
-    );
-};
 ```
-{% endraw %}
+{
+    body: {
+        errors: {
+            source: 'error message',
+        }
+    }
+}
+```
 
-**Tip**: The shape of the returned validation errors must correspond to the form: a key needs to match a `source` prop.
+**Tip**: The shape of the returned validation errors must match the form shape: each key needs to match a `source` prop.
 
 **Tip**: The returned validation errors might have any validation format we support (simple strings or object with message and args) for each key.
+
+**Tip**: If your data provider leverages our [`httpClient`](https://marmelab.com/react-admin/DataProviderWriting.html#example-rest-implementation), this will be handled automatically when your server returns an invalid response with a json body contains the `errors` key.
```

Comment:
```suggestion
**Tip**: If your data provider leverages our [`httpClient`](https://marmelab.com/react-admin/DataProviderWriting.html#example-rest-implementation), this will be handled automatically when your server returns an invalid response with a json body containing the `errors` key.
```